src/data-management/store.ts

```diff
@@ -0,0 +1,99 @@
+import { create } from 'zustand';
+import { StorageService } from './storage-service';
+import { Note } from './types';
+
+export interface NotesState {
+    notes: Note[];
+    trash: Note[];
+
+    addNote: (newNote: Pick<Note, 'title' | 'description'>) => void;
+    updateNote: (
+        id: string,
+        updateNote: Partial<Pick<Note, 'title' | 'description' | 'isPinned'>>
+    ) => void;
+    deleteNote: (id: string, permanently?: boolean) => void;
+    syncNotes: () => void;
+    syncTrash: () => void;
+    getNoteById: (id: string) => Promise<Note | undefined>;
+    restoreNote: (id: string) => Promise<void>;
+    pinNote: (id: string) => Promise<void>;
+    unPinNote: (id: string) => Promise<void>;
+}
+
+const storageService = new StorageService();
+
+export const useNoteStore = create<NotesState>((set) => {
+    const syncNotes = async () => {
+        const notes = await storageService.getAllNotes();
+        set({ notes: notes.filter((note) => !note.deletedAt) });
+    };
+
+    const syncTrash = async () => {
+        const trash = await storageService.getAllNotes();
+        set({ trash: trash.filter((note) => note.deletedAt) });
+    };
```

Comment:
In a way how it looks right now i don't see any point in having `syncNotes` and `syncTrash`.
just `syncNotes` that sets notes and removed notes should be enough